web/templates/password_reset/reset.html.eex

```diff
@@ -0,0 +1,27 @@
+<p class="lead">Enter new password</p>
+
+<%= form_for @changeset, password_reset_path(@conn, :update, @token_string), [method: :put], fn f -> %>
+  <%= if @changeset.action do %>
+    <div class="alert alert-danger">
+      <p>There was a problem resetting your password.</p>
+    </div>
+  <% end %>
+
+  <%= hidden_input f, :token, value: value_from_changeset(@changeset, :token) %>
+
+  <div class="form-group">
+    <%= label f, :password, class: "control-label" %>
+    <%= password_input f, :password, class: "form-control" %>
+    <%= error_tag f, :password %>
+  </div>
+
+  <div class="form-group">
+    <%= label f, :password_confirmation, class: "control-label" %>
+    <%= password_input f, :password_confirmation, class: "form-control" %>
+    <%= error_tag f, :password_confirmation %>
+  </div>
+
+  <div class="form-group">
+    <%= submit "Submit", class: "btn btn-primary pull-right" %>
+  </div>
+<% end %>
```

Comment:
So I thought about forcing the user to re-enter the email associated with the user, but I thought it was overkill. If the token is intercepted in some way, it's pretty likely that the attacker already has the email associated with that token. I can't think of a situation where this wouldn't be the case.

That said, doing so might be a good guard against this code malfunctioning in the future.

I think keeping it simple and leaving off the email validation is probably the correct thing to do.